core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb_test.go

```diff
@@ -1365,3 +1365,53 @@ func TestMissingRevisionRetrievalFromCache(t *testing.T) {
 	require.Equal(t, "rev1", revisions["key1"])
 	require.Equal(t, "rev2", revisions["key2"])
 }
+
+func TestGetAndUpdateNamespaceDBsInfo(t *testing.T) {
```

Comment:
We need to simulate peer failure scenarios in the test:
1. assume that the database is created and the peer fails before updating the channel metadata. When the peer comes up again and calls the `getNamespaceDBHandle()`, it should be reflected in the channel metadata.
2. a similar failure test for `newVersionedDB()` too.